sdk/scheduler/src/main/java/org/apache/mesos/offer/OfferEvaluator.java

```diff
@@ -41,47 +40,68 @@ public OfferEvaluator(StateStore stateStore) {
 
     public List<OfferRecommendation> evaluate(OfferRequirement offerRequirement, List<Offer> offers)
             throws StateStoreException {
-        Optional<PlacementRule> placementRule = getPlacementRule(offerRequirement, stateStore);
-        if (placementRule.isPresent()) {
+
+        // First, check placement constraints (to filter offers)
+        List<Offer> filteredOffers = new ArrayList<>();
+        Optional<PlacementRule> placementRuleOptional = offerRequirement.getPlacementRuleOptional();
+        if (placementRuleOptional.isPresent()) {
```

Comment:
Can we please factor out this Offer/Resource filtering into a separate method to keep `evaluate(`)'s size down?